src/pages/ConciergePage.tsx

```diff
@@ -1,36 +1,31 @@
 import {useFocusEffect} from '@react-navigation/native';
-import PropTypes from 'prop-types';
+import type {StackScreenProps} from '@react-navigation/stack';
 import React from 'react';
-import {withOnyx} from 'react-native-onyx';
-import _ from 'underscore';
+import {OnyxEntry, withOnyx} from 'react-native-onyx';
 import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
 import Navigation from '@libs/Navigation/Navigation';
+import type {AuthScreensParamList} from '@libs/Navigation/types';
 import * as Report from '@userActions/Report';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import SCREENS from '@src/SCREENS';
+import {Session} from '@src/types/onyx';
 
-const propTypes = {
+type ConciergePageOnyxProps = {
     /** Session info for the currently logged in user. */
-    session: PropTypes.shape({
-        /** Currently logged in user authToken */
-        authToken: PropTypes.string,
-    }),
+    session: OnyxEntry<Session>;
 };
 
-const defaultProps = {
-    session: {
-        authToken: null,
-    },
-};
+type ConciergePageProps = ConciergePageOnyxProps & StackScreenProps<AuthScreensParamList, typeof SCREENS.CONCIERGE>;
 
 /*
  * This is a "utility page", that does this:
  *     - If the user is authenticated, find their concierge chat and re-route to it
  *     - Else re-route to the login page
  */
-function ConciergePage(props) {
+function ConciergePage({session}: ConciergePageProps) {
     useFocusEffect(() => {
-        if (_.has(props.session, 'authToken')) {
+        if (session?.authToken) {
```

Comment:
Please make sure it doesn't break anything, cause with previous approach the expression returned true even if authToken was `null` or `undefined`

Reply:
@VickyStash You have a point! In that case, the **if-then-else** is redundant because the **then** clause is always executed since the default props had the property defined. 

`Route.Home` points to different components depending on whether the user is authenticated on not, so the Navigation.goBack(Route.Home) will always move the user into the correct page. The `Report.navigateToConciergeChat();` will navigate the user into a report but if he's not authenticated it shouldn't do anything because the component doesn't exist on the PublicStack, otherwise it will move the user to the right report, which in this case it the Concierge report.

I would remove the if-then-else and only leave the then clause, but I may be missing something here so I'd like to hear your input @VickyStash @fabioh8010 !

Reply:
`_.has` function checks if the property `authToken` inside `session` exists, even if it's `null` or `undefined` like @VickyStash stated, e.g.:

```ts
obj = {
    a: 1,
    b: null,
    c: undefined
}

_.has(obj, 'a'); // true
Boolean(obj.a); // true

_.has(obj, 'b'); // true
Boolean(obj.b); // false

_.has(obj, 'c'); // true
Boolean(obj.c); // false

_.has(obj, 'd'); // false
Boolean(obj.d); // false
```

We can have differences in some truthy/falsy values as well, e.g.:

```ts
obj2 = {
    a: {},
    b: '',
    c: 0
}

_.has(obj2, 'a'); // true
Boolean(obj2.a); // true

_.has(obj2, 'b'); // true
Boolean(obj2.b); // false because '' is a falsy value

_.has(obj2, 'c'); // true
Boolean(obj2.c); // false because 0 is a falsy value
````

@pac-guerreiro If using `if (session?.authToken) {` won't make difference for the cases mentioned above, so we are good to go with the change!